torch/_inductor/runtime/benchmarking.py

```diff
@@ -195,11 +198,189 @@ def benchmark_gpu(self: Self, _callable: Callable[[], Any], **kwargs: Any) -> fl
         this is the first requested quantile. Else, if `kwargs["return_mode"]` is specified,
         this is the requested return mode. Otherwise, this is the median.
         """
+
+        # this method may be used as a fallback if certain benchmarking features are disabled,
+        # in which case those requests may contain kwargs that are not specific to Triton's
+        # `do_bench_gpu`. as such, we may need to prune out kwargs that do not exists in
+        # `do_bench_gpu`'s signature.
+        do_bench_params = inspect.signature(self.triton_do_bench).parameters
+        for kwarg in list(kwargs.keys()):
+            if kwarg not in do_bench_params:
+                del kwargs[kwarg]
+
         if "quantiles" in kwargs:
             return self.triton_do_bench(_callable, **kwargs)[0]
         elif "return_mode" in kwargs:
             return self.triton_do_bench(_callable, **kwargs)
         return self.triton_do_bench(_callable, **kwargs, return_mode="median")
 
 
-benchmarker = TritonBenchmarker()
+def is_feature_enabled(feature_name: str) -> bool:
+    """Method to decide whether or not a feature is enabled. For more context, see the
+    benchmarking configuration section in `torch._inductor.config.benchmarking`.
+    """
+    feature_config = getattr(benchmarking_config, feature_name)
+    if feature_config.env_val is not None:
+        if feature_config.env_val == "1":
+            return True
+        elif feature_config.env_val == "0":
+            return False
+    if not is_fbcode():
+        return feature_config.oss_default
+    if feature_config.local_version is not None:
+        return (
```

Comment:
Feel like this could be standardized with some of our other tri state/ jk checking.. although I have been less involved with that. cc @masnesral  @oulgen

Reply:
Yeah - FYI - you can already do allmost of this via Config objects in the config
https://github.com/pytorch/pytorch/blob/1986b46d6349f3692c1a25ba9a3f1fcb8006751c/torch/utils/_config_module.py#L22

i.e. this is almost equivalent to benchmarking_config.feature_x = Config(env_value_default="env_name, default=oss_default, justknob-"pytorch/benchmarking:feature_name")

Note that today we don't support the version pattern (and I'm working on making JK commit time aware, so I can just kill it totally and make it a native feature of JK).

If you feel strongly about wanting to use versions in JK with config objects, I can stack a PR to add support..